common/src/main/java/net/whimxiqal/journey/data/DataManagerImpl.java

```diff
@@ -23,98 +23,88 @@
-            addBatchesToStatement("/data/sql/schema/sqlite.sql", statement);
-            statement.executeBatch();
-          } catch (SQLException e) {
-            setupSchema = false;
-            e.printStackTrace();
-          }
+
+        if (currentVersionNumber < DataVersion.latest().internalVersion) {
+          SqlMigrationHandler migrationHandler = new SqliteMigrationHandler();
+          migrationHandler.runMigration(currentVersion);
+        }
+        break;
+      case MYSQL:
+        MySqlConnectionController mysqlController = new MySqlConnectionController();
+
+        currentVersion = new MysqlDataVersionManager(mysqlController).version();
+        currentVersionNumber = currentVersion.internalVersion;
+
+        personalWaypointManager = new SqlPersonalWaypointManager(mysqlController);
+        publicWaypointManager = new SqlPublicWaypointManager(mysqlController);
+        pathRecordManager = new SqlPathRecordManager(mysqlController);
+        tunnelDataManager = new SqlTunnelDataManager(mysqlController);
+
+        if (currentVersionNumber < DataVersion.latest().internalVersion) {
+          SqlMigrationHandler migrationHandler = new MysqlMigrationHandler();
+          migrationHandler.runMigration(currentVersion);
         }
         break;
-//      case MYSQL:
-//        MySqlConnectionController mysqlController = new MySqlConnectionController();
-//        personalWaypointManager = new SqlPersonalWaypointManager(mysqlController);
-//        publicWaypointManager = new SqlPublicWaypointManager(mysqlController);
-//        pathRecordManager = new SqlPathRecordManager(mysqlController);
-//        tunnelDataManager = new SqlTunnelDataManager(mysqlController);
-//        if (setupSchema) {
-//          try (Connection connection = mysqlController.establishConnection()) {
-//            Statement statement = connection.createStatement();
-//            addBatchesToStatement("/data/sql/schema/mysql.sql", statement);
-//            statement.executeBatch();
-//          } catch (SQLException e) {
-//            setupSchema = false;
-//            e.printStackTrace();
-//          }
-//        }
-//        break;
       default:
         throw new RuntimeException();
     }
-    if (setupSchema) {
-      DataVersion.writeVersion(DataVersion.latest());
-    }
-  }
 
-  private void addBatchesToStatement(String queryResource, Statement statement) throws SQLException {
-    InputStream resourceStream = this.getClass().getResourceAsStream(queryResource);
-    if (resourceStream == null) {
-      throw new NoSuchElementException("Cannot find resource at path " + queryResource);
-    }
-    for (String query : new BufferedReader(new InputStreamReader(resourceStream))
-        .lines().collect(Collectors.joining("\n")).split(";")) {
-      statement.addBatch(query);
-    }
+    // The database should always be on the latest version at this point, or have errored out before here.
+    databaseVersion = DataVersion.latest();
```

Comment:
I think it makes sense for the migration handler to return whatever database version it was updated to, and then set `databaseVersion` in the `DataManager` here to be whatever the migration handler returned. Setting it manually here ignores the result of that method, even though we know that the current code of `runMigration` does in fact return `latest()` every time.